shakenfist/artifact.py

```diff
@@ -123,6 +123,9 @@ def add_index(self, blob_uuid):
             self.log.with_fields(entry).info('Added index to artifact')
             return entry
 
+    def del_index(self, index):
```

Comment:
I hadn't thought this through. How do we handle not reusing an index? Or does it not matter if we reuse an index? It would have to be the highest index for it to be reused. Should we instead somehow mark indexes as deleted but leave them in the database? That will bloat the database over time. Discuss.

Reply:
Yeah, that seems fair. I think a "highest index" as an attribute of the artifact would also save the potentially expensive index scan we do here too.

Reply:
https://github.com/shakenfist/shakenfist/issues/1060